partner_communication_switzerland/models/mail_tracking_event.py

```diff
@@ -38,6 +38,56 @@ def process_delivered(self, tracking_email, metadata):
         return super(MailTrackingEvent, self).process_delivered(
             tracking_email, metadata)
 
+    @api.model
+    def process_hard_bounce(self, tracking_email, metadata):
+        partner = tracking_email.partner_id
+        if partner:
+            self._invalid_email(tracking_email)
```

Comment:
All this code is duplicated three times. Could you refactor that into a function please?